File: scripts/rewrite_document_id.py
#!/usr/bin/env python3
import os
import os.path
import sys
import json

DIR = sys.argv[1]

def main():
    for (dirpath, dirnames, filenames) in os.walk(DIR):
        for filename in filenames:
            filepath = os.path.join(dirpath, filename)
            relpath = os.path.relpath(filepath, DIR)
            relpath_noext, ext = os.path.splitext(relpath)
            # Ignore non-md files
            if ext != ".md":
                continue
            with open(filepath, "r") as f:
                data = f.read()
            lines = [line for line in data.split("\n") if not line.startswith("id: ")]
            with open(filepath, "w") as f:
                for line in lines:
                    print(line, file=f)


if __name__ == "__main__":
    main()

File: scripts/rewrite_document_id.js
const fs = require("fs");
const path = require("path");

const DIR = process.argv[2];

function recur(dir) {
  const entries = fs.readdirSync(dir, {
    encoding: "utf8",
    withFileTypes: true,
  });
  for (const entry of entries) {
    const filepath = path.join(dir, entry.name);
    if (entry.isDirectory()) {
      recur(filepath);
    } else if (entry.isFile()) {
      const relpath = path.relative(DIR, filepath);
      const ext = path.extname(relpath);
      // Ignore non-md files
      if (ext !== ".md") {
        continue;
      }

      let content = fs.readFileSync(filepath, {
        encoding: "utf8",
        flag: "r",
      });
      let lines = content.split("\n");
      lines = lines.filter((line) => !line.startsWith("id: "));
      content = lines.join("\n") + "\n";
      fs.writeFileSync(filepath, content);
    }
  }
}

recur(DIR);
